host-configs/darwin-clang.cmake

```diff
@@ -11,20 +11,26 @@ set(MPI_C_COMPILER "/usr/local/bin/mpicc" CACHE PATH "")
 set(MPI_CXX_COMPILER "/usr/local/bin/mpicxx" CACHE PATH "")
 set(MPIEXEC "/usr/local/bin/mpirun" CACHE PATH "")
 
-set(ENABLE_GTEST_DEATH_TESTS ON CACHE BOOL "" FORCE )
+set(ENABLE_GTEST_DEATH_TESTS ON CACHE BOOL "" FORCE)
 
 set(ENABLE_PAMELA ON CACHE BOOL "" FORCE)
 set(ENABLE_PVTPackage ON CACHE BOOL "" FORCE)
 set(ENABLE_GEOSX_PTP ON CACHE BOOL "" FORCE)
 
-set(CUDA_ENABLED      "OFF"       CACHE PATH "" FORCE)
-set(ENABLE_OPENMP     "OFF"        CACHE PATH "" FORCE)
+set(CUDA_ENABLED "OFF" CACHE PATH "" FORCE)
+set(ENABLE_OPENMP "OFF" CACHE PATH "" FORCE)
 
-option( ENABLE_CALIPER "Enables CALIPER" OFF )
+option(ENABLE_CALIPER "Enables CALIPER" OFF)
 
-set( BLAS_LIBRARIES /usr/lib/libblas.dylib CACHE PATH "" FORCE )
-set( LAPACK_LIBRARIES /usr/lib/liblapack.dylib CACHE PATH "" FORCE )
+set(BLAS_LIBRARIES /usr/lib/libblas.dylib CACHE PATH "" FORCE)
+set(LAPACK_LIBRARIES /usr/lib/liblapack.dylib CACHE PATH "" FORCE )
+
+set(ENABLE_DOXYGEN OFF CACHE BOOL "" FORCE)
 
-set( ENABLE_DOXYGEN OFF CACHE BOOL "" FORCE )
 #set( DOXYGEN_EXECUTABLE /usr/local/bin/doxygen CACHE PATH "" FORCE )
-#set( SPHINX_EXECUTABLE /usr/local/bin/sphinx-build CACHE PATH "" FORCE )
\ No newline at end of file
+#set( SPHINX_EXECUTABLE /usr/local/bin/sphinx-build CACHE PATH "" FORCE )
+
+set(ENABLE_GEOSX_PTP ON CACHE BOOL "" FORCE)
+
+set(GEOSX_TPL_DIR "/usr/local/GEOSX/GEOSX_TPL" CACHE PATH "" FORCE)
```

Comment:
The value of `GEOSX_TPL_DIR` is actually provided as a meta-information of the GCP blob.
Then it is provided from `.travis.yml` as a parameter (`-DGEOSX_TPL_DIR=...`).
So it should not be needed to hardcode it here.

If you want to use it as an env variable as for other docker linux cases, you should probably define the `env` variable directly inside the `.travis/yml` and call `ENV{GEOSX_TPL_DIR}` (currently, `GEOSX_TPL_DIR` is just a variable, not an `env` variable).

Just letting the old definition in `.travis.yml` and nothing in the current `darwin-clang.cmake` seems easier.

Reply:
When we move to spack we'll have separate host configs for each docker image (which I personally like a lot better than all the environment variables). The way I have it here is that `GEOSX_TPL_DIR` needs to be defined when processing `tpls.cmake`, and it appears that variables defined with `-D` don't get defined until after processing the config file.

In the `darwin-clang.cmake` config I tried to do `ENV{GEOSX_TPL_DIR}` but it wasn't defined, I even added in an `echo $GEOSX_TPL_DIR` before the call to CMake to make sure it was actually defined in the environment.